app/Http/Controllers/PostController.php

```diff
@@ -27,26 +27,34 @@ public function __construct()
     public function show($id)
     {
         $data = []; //to be sent to the view
-
-        $post = Post::findOrFail($id);
+        $post = Post::with("comments")->find($id);
         $data["title"] = $post->getTitle();
         $data["post"] = $post;
+        if ($post->user == Auth::user()) {
+            $data["allowed_ops"] = true;
+        } else {
+            $data["allowed_ops"] = false;
+        }
         return view('forum.show')->with("data", $data);
     }
 
     /**
-     * Delete a post along with the associated comments
+     * Delete a post along with the associated comments. Users only can delete
+     * their comments.
      * 
      * @return \Illuminate\Http\Response
      */
-    public function destroy($id){
+    public function delete($id)
+    {
         $post = Post::findOrFail($id);
-        $post->comments()->delete();
-        $post->delete();
-        $data["title"] = "Posts list";
-        $data["posts"] = Post::all();
 
-        return view('forum.index')->with("data", $data, 'success', 'Post deleted successfully!');
+        if ($post->user == Auth::user()) {
+            $post->comments()->delete();
+            $post->delete();
```

Comment:
If database is configured in cascade mode, there's no need to specify the deletion of post's comments. Imagine there were more tables attached to the post. Would you delete each one in these lines? Remove the line with post->comments->delete and ensure database is set up on cascade mode

Reply:
It isn't. For that reason I need this way.

Reply:
I think that it doesn't  matter now.

Reply:
Thats the point. It'll matter later.